MonoGame.Framework/Audio/OALSoundBuffer.cs

```diff
@@ -56,6 +56,29 @@ public void BindDataBuffer(byte[] dataBuffer, ALFormat format, int size, int sam
                 ALHelper.CheckError("Failed to fill buffer.");
             }
 
+            bool formatLoopEnabled = true;
+
+            /* Formats that OpenAL Soft currently doesn't support loop points for:
+             * MonoIma4
+             * StereoIma4
+             * MonoMSAdpcm
+             * StereoMSAdpcm
+            */
+            if (format == ALFormat.MonoIma4 || format == ALFormat.StereoIma4
+                || format == ALFormat.MonoMSAdpcm || format == ALFormat.StereoMSAdpcm)
+            {
+                formatLoopEnabled = false;
+            }
+
+            if (OpenALSoundController.GetInstance.SupportsLoopPoints && formatLoopEnabled && loopStart >= 0 && loopLength > 0)
+            {
+                //Loop end is loopStart + loopLength
+                int[] loopData = new int[2] { loopStart, loopStart + loopLength };
```

Comment:
@tdeeb Could you do a `stackalloc` here? The function will need to be marked `unsafe`, but that's fine.